src/main/java/io/cdap/plugin/successfactors/source/config/SuccessFactorsPluginConfig.java

```diff
@@ -0,0 +1,371 @@
+  }
+
+  @Nullable
+  public String getFilterOption() {
+    // Plugin UI field is 'textarea' so the user can input multiline filter statement
+    // and as line break are not supported in URI
+    // so any line break is removed from the filter option.
+    return SuccessFactorsUtil.removeLinebreak(this.filterOption);
+  }
+
+  @Nullable
+  public String getSelectOption() {
+    // Plugin UI field is 'textarea' so the user can input multiline select statement
+    // and as line break are not supported in URI and any select column name with extra
+    // spaces are considered as actual column by SuccessFactors services and results in 'Not Found'
+    // so to avoid the 'Not Found' scenarios
+    // any line break and extra spaces is removed from the select option.
+    // e.g.
+    //    $select = col1, col2,
+    //                col3,col4
+    //    will be convert to col1,col2,col3,col4
+    return SuccessFactorsUtil.removeWhitespace(this.selectOption);
+  }
+
+  @Nullable
+  public String getExpandOption() {
+    return SuccessFactorsUtil.removeWhitespace(this.expandOption);
+  }
+
+  public String getPaginationType() {
+    return this.paginationType;
+  }
+
+  /**
+   * Checks if the call to SuccessFactors service is required for metadata creation.
+   * condition parameters: ['host' | 'serviceName' | 'entityName' | 'username' | 'password']
+   * - any parameter is 'macro' then it returns 'false'
+   *
+   * @return boolean flag as per the check
+   */
+  public boolean isSchemaBuildRequired() {
+    return !(containsMacro(UNAME) || containsMacro(PASSWORD) || containsMacro(BASE_URL) || containsMacro(ENTITY_NAME));
+  }
+
+  /**
+   * @return the schema of the dataset
+   */
+  @Nullable
+  public Schema getSchema(FailureCollector collector) {
+    try {
+      return Strings.isNullOrEmpty(schema) ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      collector.addFailure("Invalid schema: " + e.getMessage(), null)
+        .withConfigProperty(NAME_SCHEMA);
+    }
+    // if there was an error that was added, it will throw an exception, otherwise, this statement will not be executed
+    throw collector.getOrThrowException();
+  }
+
+  public void validatePluginParameters(FailureCollector failureCollector) {
+    validateMandatoryParameters(failureCollector);
+    validateBasicCredentials(failureCollector);
+    LOG.debug("Validating the advanced parameters.");
+    validateEntityParameter(failureCollector);
+    failureCollector.getOrThrowException();
+  }
+
+  /**
+   * Validates the mandatory parameters.
+   *
+   * @param failureCollector {@code FailureCollector}
+   */
+  private void validateMandatoryParameters(FailureCollector failureCollector) {
+
+    IdUtils.validateReferenceName(getReferenceName(), failureCollector);
+    if (SuccessFactorsUtil.isNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("SAP SuccessFactor Base URL");
```

Comment:
use variable for SAP SuccessFactor Base URL

Reply:
Added